src/network/networkctl.c

```diff
@@ -1393,89 +1393,74 @@ static int dump_lldp_neighbors(Table *table, const char *prefix, int ifindex) {
-                uint32_t family;
+                return r;
 
-                r = sd_bus_message_read(reply, "u", &family);
-                if (r < 0)
-                        return bus_log_parse_error(r);
+        JSON_VARIANT_ARRAY_FOREACH(i, json_variant_by_key(v, "Interfaces")) {
+                JsonVariant *dhcp_server = NULL, *lease = NULL;
 
-                r = sd_bus_message_read_array(reply, 'y', &client_id, &client_id_sz);
-                if (r < 0)
-                        return bus_log_parse_error(r);
+                if (json_variant_integer(json_variant_by_key(i, "Index")) != link->ifindex)
+                        continue;
 
-                r = sd_bus_message_read_array(reply, 'y', &addr, &sz);
-                if (r < 0 || sz != 4)
-                        return bus_log_parse_error(r);
+                dhcp_server = json_variant_by_key(i, "DHCPServer");
+                if (json_variant_is_null(dhcp_server))
+                        continue;
 
-                r = sd_bus_message_read_array(reply, 'y', &gtw, &sz);
-                if (r < 0 || sz != 4)
-                        return bus_log_parse_error(r);
+                JSON_VARIANT_ARRAY_FOREACH(lease, json_variant_by_key(dhcp_server, "Leases")) {
+                        _cleanup_free_ uint8_t *client_id = NULL, *addr = NULL;
+                        _cleanup_free_ char *hostname = NULL, *id = NULL, *ip = NULL;
+                        size_t client_id_sz, addr_sz;
+                        uint64_t expiration_usec;
 
-                r = sd_bus_message_read_array(reply, 'y', &hwaddr, &sz);
-                if (r < 0)
-                        return bus_log_parse_error(r);
+                        r = json_variant_bytes(
+                                        json_variant_by_key(lease, "ClientId"), &client_id, &client_id_sz);
+                        if (r < 0)
+                                return r;
+                        r = sd_dhcp_client_id_to_string((void *) client_id, client_id_sz, &id);
+                        if (r < 0)
+                                return r;
 
-                r = sd_bus_message_read_basic(reply, 't', &expiration);
-                if (r < 0)
-                        return bus_log_parse_error(r);
+                        r = json_variant_bytes(json_variant_by_key(lease, "Address"), &addr, &addr_sz);
+                        if (r < 0)
+                                return r;
+                        if (addr_sz != sizeof(in_addr_t))
+                                return -EINVAL;
+                        r = in_addr_to_string(AF_INET, (void *) addr, &ip);
+                        if (r < 0)
+                                return r;
 
-                r = sd_dhcp_client_id_to_string(client_id, client_id_sz, &id);
-                if (r < 0)
-                        return bus_log_parse_error(r);
+                        if (json_variant_by_key(lease, "Hostname"))
+                                asprintf(&hostname,
```

Comment:
Missing error handling.
I do not think it is necessary to format the string. Please do that in the below.

Reply:
>Missing error handling.

Done.

> I do not think it is necessary to format the string. Please do that in the below.

I am wrapping the hostname in `"` here and conditionally adding the additional `,`. Not sure how I would do this below when hostname might not be set at all (see example above. One of my devices does not send the hostname at all).